core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -905,15 +905,8 @@ private[spark] object MapOutputTracker extends Logging {
         for (part <- startPartition until endPartition) {
           val size = status.getSizeForBlock(part)
           if (size != 0) {
-            if (useOldFetchProtocol) {
```

Comment:
Thanks for the report and fix! The root cause here is while we set `useOldFetchProtocol` here, the shuffle id in the reader side and the writer side are inconsistency.
But we can't fix like this, because while `useOldFetchProtocl=false`, we'll use the old version of fetching protocol `OpenBlocks`, which consider map id is int and will directly parse the string into Integer. See here: https://github.com/apache/spark/blob/148cd26799c69ab9cfdc2b3b8000a194c12518b8/common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java#L296

So the right way I think is doing the fix in `ShuffleWriteProcessor`, we should fill mapId with `mapTaskId` or `mapIndex` denpending on config `spark.shuffle.useOldFetchProtocol`.

Reply:
sorry, could you explain why  Integer directly parse the string for the big and long-running application not work? 
Is it a performance problem?
looking forward for your reply.